src/core/chunked_stream.js

```diff
@@ -267,24 +245,489 @@ var ChunkedStream = (function ChunkedStreamClosure() {
         return missingChunks;
       };
       var subStream = new ChunkedStreamSubstream();
+      subStream.ensureRange(start, start + length);
       subStream.pos = subStream.start = start;
       subStream.end = start + length || this.end;
       subStream.dict = dict;
+      subStream.totalLength = subStream.end - subStream.start;
+      subStream.subStream = true;
+
+      if (subStream.start >= subStream.buffer.start &&
+          subStream.end <= subStream.buffer.end) {
+        // Stream is fully loaded, use fast getByteMethod
+        subStream.getByte = this.continuousAlloc ?
+          this.getByteFastContinuous(subStream.buffer.buffer, subStream.end) :
+          this.getByteFast;
+      }
       return subStream;
     },
 
     isStream: true
   };
 
-  return ChunkedStream;
+  return ChunkedStreamBase;
+})();
+
+/* Chunked stream implementation that allocates all memory required for PDF in
+ single continuous array. Better performing than its Fragmented version but
+ not suitable for larger PDFs  as it will run out of memory. E.g. PDF size
+ of 200MB will allocate 200MB even if no page  has been loaded yet.
+ */
+
+var ChunkedStreamContinuous = (function ChunkedStreamContinuousClosure() {
+
+  function ChunkedStreamContinuous(length, chunkSize, manager) {
+    ChunkedStreamBase.call(this, length, chunkSize, manager);
+      this.continuousAlloc = true;
```

Comment:
The indenting is off here. These lines all need to be aligned with `ChunkedStreamBase` above.